pkg/ipam-node/allocator/range_set.go

```diff
@@ -0,0 +1,99 @@
+/*
+ Copyright 2023, NVIDIA CORPORATION & AFFILIATES
+ Copyright 2015 CNI authors
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+     http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package allocator
+
+import (
+	"fmt"
+	"net"
+	"strings"
+)
+
+type RangeSet []Range
```

Comment:
Technically we dont really need rangeset for now.

That said, im ok if you want to keep it.

Reply:
let's keep it for now, we may need this logic in the future